includes/class-twitter-accounts.php

```diff
@@ -0,0 +1,331 @@
+	}
+
+	/**
+	 * Authorize the user with Twitter.
+	 */
+	public function twitter_authorize() {
+		// Check if the user has the correct permissions.
+		if ( ! current_user_can( 'manage_options' ) ) {
+			wp_die( esc_html__( 'You do not have sufficient permissions to access this page.', 'autoshare-for-twitter' ) );
+		}
+
+		// Check if the nonce is valid.
+		if ( ! isset( $_GET['autoshare_twitter_authorize_nonce'] ) || ! wp_verify_nonce( sanitize_text_field( wp_unslash( $_GET['autoshare_twitter_authorize_nonce'] ) ), 'autoshare_twitter_authorize_action' ) ) {
+			wp_die( esc_html__( 'You have not access to doing this operations.', 'autoshare-for-twitter' ) );
+		}
+
+		// Get the request token.
+		$callback_url = admin_url( 'admin-post.php?action=authoshare_authorize_callback' );
+
+		try {
+			$request_token = $this->twitter_api->request_token( $callback_url );
+
+			// Save temporary credentials to cookies for later use.
+			setcookie( 'autoshare_oauth_token', $request_token['oauth_token'], time() + 3600, '/' );
+			setcookie( 'autoshare_oauth_token_secret', $request_token['oauth_token_secret'], time() + 3600, '/' );
+
+			// Initiate authorization.
+			$url = $this->twitter_api->get_authorize_url( $request_token['oauth_token'] );
+			if ( ! empty( $url ) ) {
+				wp_redirect( $url ); // phpcs:ignore WordPress.Security.SafeRedirect.wp_redirect_wp_redirect
+				exit();
+			}
+		} catch ( \Exception $e ) {
+			$error   = $e->getMessage();
+			$decoded = json_decode( $e->getMessage() );
+			if ( json_last_error() === JSON_ERROR_NONE ) {
+				$error = $decoded;
+			}
+
+			if ( ! empty( $error->errors ) ) {
+				$error         = current( $error->errors );
+				$error_message = $error->message;
+			} elseif ( ! empty( $error ) ) {
+				$error_message = $error;
+			} else {
+				$error_message = __( 'Something went wrong. Please try again.', 'autoshare-for-twitter' );
+			}
+
+			$this->set_connection_notice( 'error', $error_message );
+		}
+
+		// Redirect back to AutoShare settings page.
+		wp_safe_redirect( admin_url( 'options-general.php?page=autoshare-for-twitter' ) );
+		exit();
+	}
+
+	/**
+	 * Callback for Twitter authorization process.
+	 *
+	 * @throws \Exception If the request token is not valid.
+	 */
+	public function authoshare_authorize_user_callback() {
+		// Check if the user has the correct permissions.
+		if ( ! current_user_can( 'manage_options' ) ) {
+			wp_die( esc_html__( 'You do not have sufficient permissions to access this page.', 'autoshare-for-twitter' ) );
+		}
+
+		try {
+			// Check if users has denied the authorization.
+			// phpcs:ignore WordPress.Security.NonceVerification.Recommended
+			$is_deniend = isset( $_GET['denied'] ) ? sanitize_text_field( wp_unslash( $_GET['denied'] ) ) : false;
+			if ( $is_deniend ) {
+				throw new \Exception( __( 'You have denied the authorization.', 'autoshare-for-twitter' ) );
```

Comment:
@iamdharmesh Correct me if I'm wrong, by the code it seems that the API returns with this error.
This can be "Authorization denied for this request."